factory.py

```diff
@@ -131,10 +158,15 @@ def algorithm(self) -> Algorithm:
     def algorithm(self, instance: Algorithm) -> None:
         self._instance = instance
 
-    def set_logger(self, path: Union[str, Path]) -> None:
-        set_logger(logging_path=path)
+    def set_logger(
+        self, path: Union[str, Path], stream_stdout: Optional[bool] = None
+    ) -> None:
+        set_logger(logging_path=path, stream_stdout=stream_stdout)
 
     def run(self) -> None:
+        """Overloaded method for each Job Service"""
         if self.algorithm is None:
             raise ValueError("No algorithm instance has been created.")
-        self.algorithm.compress_model()
+
+        if self._type != FactoryTypes.ADAPT:
+            self.algorithm.compress_model()
```

Comment:
Because this is not being overloaded in KOMPRESS. It will require changes in KOMPRESS again and to ensure that didn't affect KOMPRESS workflow, this change was made.

Reply:
uhh? how would rewriting this method in adapt break kompress changes? and this method is not implemented in vision or text-gen maybe because the current implementation worked out there.

Reply:
can u also share the error trace here for a better understanding on the breaking changes.

Reply:
Re writing this method in ADAPt will not break KOMPRESS. What I am saying is we have 1 mthod in the parent factory which runs kompress only. Why have a separate method in adapt to run that? That will create a non uniformity in the code right ? It is not like this run mehod is being over written in text-gen, vision factories too right ? We can have one factory to call all 3 methods for the sake of uniformity. The only reason the parent factory has compress_model() method is that maybe it was written with only kompress in mind. 

If it was to be overwritten in individual factories, then that design should be maintained across all 3 repos to maintain uniformity.

Reply:
> Why have a separate method in adapt to run that?

That's subclassing and overwriting.

> That will create a non uniformity in the code right?

Not quite. such changes are what we build abstraction classes for; to subclass and modify, extend, or change parts of a functionality while consolidating as much of the common functionalities as possible.

> It is not like this run mehod is being over written in text-gen, vision factories too right ?

That should be fine because when the repo got into structure it served only two modules and the then sensible abstraction for the run method was the same for the both.

> We can have one factory to call all 3 methods for the sake of uniformity.

If u notice, an object of nyuntam.factory is never created. We only create the vision, textgen, or adapt factory using a class_method. besides, simply creating multiple classes without any common functionalities doesn't serve much purpose (this, when the functionalities are more or less common). 


inheriting from a class and rewriting a method doesn't add a 'separate method'; when we overwrite - its just modifying the working of the method for the purpose of the subclass. we need not always use the same methods as implemented in the base class but we still do so because the base class many-a-times are responsible to set the flow of execution even if the subsequent flows are implemented by subclasses. one of the many uses of inheritance is to abstract out common functionalities and separation of logic, and that separation of logic is what overwriting enables us to do (along with other use cases); 
and no the methods weren't created with only kompress in mind, and that reflects in the vision and text-gen factory's, but rather the ability to inherit and adapt to different services. yes the naming convention are slightly more suitable for kompress - but those are the changes that is expected in the integration pr. some of the expected changes would be - 
1. renaming the default run method for Algorithm abstract class to `run` and not `compress_model` and that extended also in adapt, text-gen and vision
2. the get_factory method to use both job service and task to infer the factory type
3. the base factory pre-init to assert the get_algorithm and factory type, which is called with super() in the subclass if other fail fast assertions are added, etc.

Reply:
> 1. renaming the default run method for Algorithm abstract class to `run` and not `compress_model` and that extended also in adapt, text-gen and vision


That will require changing the code from text-gen and vision which I believe is byond the scope for this PR. Those changes should be brought in with a different PR from text-gen and vision side. 


> 2. the get_factory method to use both job service and task to infer the factory type

The get_factory does use both job_service and task to get the factories as of now. I don't understand the issue.

Reply:
changed

Reply:
If this was meant to be a parent method, then it should have been just a abract method here. It should have been always over-written in each individual factory. Doesn't make sense for it to run on 2 Factories while being needed to be overwritten for all other factories.